keepassxc-browser/background/keepass.js

```diff
@@ -332,41 +189,26 @@ keepass.generatePassword = async function(tab) {
 
         let password;
         const kpAction = kpActions.GENERATE_PASSWORD;
-        const [ nonce, incrementedNonce ] = keepass.getNonces();
+        const [ nonce, incrementedNonce ] = keepassClient.getNonces();
 
-        const request = {
+        const messageData = {
             action: kpAction,
             nonce: nonce,
             clientID: keepass.clientID
         };
 
-        const response = await keepass.sendNativeMessage(request);
-        if (response.message && response.nonce) {
-            const res = keepass.decrypt(response.message, response.nonce);
-            if (!res) {
-                keepass.handleError(tab, kpErrors.CANNOT_DECRYPT_MESSAGE);
-                return [];
-            }
-
-            const message = nacl.util.encodeUTF8(res);
-            const parsed = JSON.parse(message);
-            keepass.setcurrentKeePassXCVersion(parsed.version);
-
-            if (keepass.verifyResponse(parsed, incrementedNonce)) {
-                password = parsed.entries ?? parsed.password;
-                keepass.updateLastUsed(keepass.databaseHash);
-            } else {
-                console.log('GeneratePassword rejected');
-            }
-
-            return password;
-        } else if (response.error && response.errorCode) {
-            keepass.handleError(tab, response.errorCode, response.error);
+        const response = await keepassClient.sendMessage(kpAction, messageData, nonce);
+        const [ res, parsed ] = keepassClient.handleResponse(response, incrementedNonce, tab);
```

Comment:
As above, if parsed is `undefined` then it failed. There isn't a need for a separate res boolean.